### PR TITLE
ci: Use cargo-nextest from within a pixi task

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Also remove `test_environment_json` since it doesn't work when run from within pixi

### CODE REVIEW

**Comment by tdejager** [comment]
I dont think you should remove the rust cache, that is not cached in a pixi environment right?

**Comment by Hofer-Julian** [comment]
> I dont think you should remove the rust cache, that is not cached in a pixi environment right?

At least the target directory is part of .pixi and therefore cached. Does the rust cache action cache more than that?

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -128,35 +128,12 @@ jobs:
...
+          cache: true
 
       - name: "Cargo nextest"
-        run: >-
-          cargo nextest run
-          --workspace
-          --retries 2
-          --no-default-features --features rustls-tls --features slow_integration_tests
-          --status-level skip --failure-output immediate-final --no-fail-fast --final-status-level slow
+        run: pixi run test-all
```
Comment (ruben-arts) [inline .github/workflows/rust.yml:136]: The compile step now takes 2 min 44 seconds, where in main it takes 1 min 29 seconds. Do you have an idea why that would be?

**Review by ruben-arts — APPROVED** [review]

File: tests/install_tests.rs [hunk 2]
```diff
@@ -560,25 +558,6 @@ async fn test_installer_name() {
...
-    ))
-    .unwrap();
-    pixi::environment::update_prefix(&project.default_environment(), LockFileUsage::Update, false)
-        .await
-        .unwrap();
-    assert_eq!(
-        lock_str,
-        std::fs::read_to_string("tests/satisfiability/old_lock_file/pixi.lock").unwrap()
-    );
-}
```
[the remark this answers is not in the document]
Comment (Hofer-Julian) [reply tests/install_tests.rs]: You are right. I reverted this.

**Review by ruben-arts — APPROVED** [review]

**Pull request merged**